### DISCUSSION TITLE
Admob IDs Obfuscation

### DISCUSSION THREAD

**Discussion by DonZheng** [admob-plus/admob-plus#299] [Ideas]
At the moment, when we add adunit IDs and build Android apps, anyone gets the apk file can easily decompile and get hold of the admob IDs.

This causes critical concerns that malicious people can replace developer's adunit IDs within the apk.

Wish admob-plus can provide a handy method to obfuscate the key information.

**Comment by hyl31279** [comment]
关于你之前在shadowsocks的问题想请教一下你解决了吗

**Comment by ratson** [comment]
@DonZheng There is no way to secure Ad Unit ID as long as Google AdMob SDK requires it.
If you really need that, you can do the obfuscation when calling initializing the ad instance.
